### PR TITLE
Update adventure

### PR DESCRIPTION
**Pull request by Owen1212055**
Unsure if we should have deleteMessage implemented either, but this is a baseline and contains no signed message sending support.

### CODE REVIEW

File: patches/server/0009-Adventure.patch [hunk 1]
```diff
@@ -3530,12 +3572,18 @@ index d2be98416529aea3bdbedd0ea7131bd8de2a0162..e12b4ec0d14901a0246cfea16c65fb04
 +
 +    @Override
 +    public void playSound(final net.kyori.adventure.sound.Sound sound, final double x, final double y, final double z) {
++        final long seed;
++        if (sound.seed().isPresent()) {
++            seed = sound.seed().getAsLong();
++        } else {
++            seed = this.getHandle().getRandom().nextLong();
++        }
```
Comment (Machine-Maker) [inline patches/server/0009-Adventure.patch]: Probably should use the one-liner for all this

File: patches/server/0009-Adventure.patch [hunk 2]
```diff
@@ -3549,13 +3597,19 @@ index d2be98416529aea3bdbedd0ea7131bd8de2a0162..e12b4ec0d14901a0246cfea16c65fb04
 +        } else {
 +            throw new IllegalArgumentException("Sound emitter must be an Entity or self(), but was: " + emitter);
 +        }
++        final long seed;
++        if (sound.seed().isPresent()) {
++            seed = sound.seed().getAsLong();
++        } else {
++            seed = this.getHandle().getRandom().nextLong();
++        }
```
Comment (Machine-Maker) [inline patches/server/0009-Adventure.patch]: same as above

File: patches/server/0009-Adventure.patch [hunk 3]
```diff
@@ -3530,12 +3572,18 @@ index d2be98416529aea3bdbedd0ea7131bd8de2a0162..e12b4ec0d14901a0246cfea16c65fb04
...
 +        if (event.isPresent()) {
-+            this.getHandle().connection.send(new ClientboundSoundEntityPacket(BuiltInRegistries.SOUND_EVENT.wrapAsHolder(event.get()), io.papermc.paper.adventure.PaperAdventure.asVanilla(sound.source()), entity, sound.volume(), sound.pitch(), this.getHandle().getRandom().nextLong())); // TODO adventure sound seed
++            this.getHandle().connection.send(new ClientboundSoundEntityPacket(BuiltInRegistries.SOUND_EVENT.wrapAsHolder(event.get()), io.papermc.paper.adventure.PaperAdventure.asVanilla(sound.source()), entity, sound.volume(), sound.pitch(), seed));
 +        } else {
-+            this.getHandle().connection.send(new ClientboundSoundPacket(Holder.direct(SoundEvent.createVariableRangeEvent(name)), io.papermc.paper.adventure.PaperAdventure.asVanilla(sound.source()), x, y, z, sound.volume(), sound.pitch(), this.getHandle().getRandom().nextLong())); // TODO adventure sound seed
++            this.getHandle().connection.send(new ClientboundSoundPacket(Holder.direct(SoundEvent.createVariableRangeEvent(name)), io.papermc.paper.adventure.PaperAdventure.asVanilla(sound.source()), x, y, z, sound.volume(), sound.pitch(), seed));
 +        }
```
Comment (Machine-Maker) [inline patches/server/0009-Adventure.patch]: Why is this using the sound entity packet in the first place? That looks wrong but if its right, it supports custom sounds

File: patches/server/0009-Adventure.patch [hunk 4]
```diff
@@ -3549,13 +3597,19 @@ index d2be98416529aea3bdbedd0ea7131bd8de2a0162..e12b4ec0d14901a0246cfea16c65fb04
...
++            this.getHandle().connection.send(new net.minecraft.network.protocol.game.ClientboundSoundEntityPacket(BuiltInRegistries.SOUND_EVENT.wrapAsHolder(event.get()), io.papermc.paper.adventure.PaperAdventure.asVanilla(sound.source()), entity, sound.volume(), sound.pitch(), seed));
 +        } else {
-+            this.getHandle().connection.send(new ClientboundSoundPacket(Holder.direct(SoundEvent.createVariableRangeEvent(name)), io.papermc.paper.adventure.PaperAdventure.asVanilla(sound.source()), entity.position().x(), entity.position().y(), entity.position().z(), sound.volume(), sound.pitch(), this.getHandle().getRandom().nextLong())); // TODO adventure sound seed
++            this.getHandle().connection.send(new ClientboundSoundPacket(Holder.direct(SoundEvent.createVariableRangeEvent(name)), io.papermc.paper.adventure.PaperAdventure.asVanilla(sound.source()), entity.position().x(), entity.position().y(), entity.position().z(), sound.volume(), sound.pitch(), seed));
 +        }
```
Comment (Machine-Maker) [inline patches/server/0009-Adventure.patch]: the entity sound packet supports custom sounds now

File: patches/server/0009-Adventure.patch [hunk 5]
```diff
@@ -3422,6 +3422,48 @@ index d2be98416529aea3bdbedd0ea7131bd8de2a0162..e12b4ec0d14901a0246cfea16c65fb04
...
++//        net.minecraft.network.chat.PlayerChatMessage playerChatMessage = new net.minecraft.network.chat.PlayerChatMessage(
++//            null, // TODO:
++//            new net.minecraft.network.chat.MessageSignature(signedMessage.signature().bytes()),
++//            null, // TODO
++//            io.papermc.paper.adventure.PaperAdventure.asVanilla(signedMessage.unsignedContent()),
++//            net.minecraft.network.chat.FilterMask.PASS_THROUGH
++//        );
++//
++//        this.getHandle().sendChatMessage(net.minecraft.network.chat.OutgoingChatMessage.create(playerChatMessage), this.getHandle().isTextFilteringEnabled(), this.toHandle(boundChatType));
++//    }
```
Comment (Machine-Maker) [inline patches/server/0009-Adventure.patch]: If we aren't going to do the main impl yet, we should at least override the default impl and point system signed messages to the method that takes a bound chat type. See the default Audience#sendMessage(SignedMessage,ChatType.Bound) impl

**Review by Machine-Maker — CHANGES_REQUESTED** [review]

File: patches/server/0009-Adventure.patch (hunk 4 above)
Comment (Owen1212055) [reply patches/server/0009-Adventure.patch]: Ahh. I thought someone already made the change lol, similar names.

File: patches/server/0009-Adventure.patch [hunk 6]
```diff
@@ -3422,6 +3422,54 @@ index d2be98416529aea3bdbedd0ea7131bd8de2a0162..e12b4ec0d14901a0246cfea16c65fb04
 +    }
 +
 +    @Override
++    public void deleteMessage(net.kyori.adventure.chat.SignedMessage.Signature signature) {
++        if (getHandle().connection == null) return;
++        net.minecraft.network.chat.MessageSignature sig = new net.minecraft.network.chat.MessageSignature(signature.bytes());
++
++        this.getHandle().connection.send(new net.minecraft.network.protocol.game.ClientboundDeleteChatPacket(new net.minecraft.network.chat.MessageSignature.Packed(0, sig)));
```
Comment (Machine-Maker) [inline patches/server/0009-Adventure.patch]: 0 is not the "ignore" message id. See the other ctor for MessageSignature.Packed which uses -1. Just use that other ctor that only takes the MessageSignature object.

**Pull request merged**